## Import libraries and read in list of movie IDs

In [1]:
import imdb
import pandas
import time
import numpy as np

In [7]:
movies_and_ids = pandas.read_csv("CS109B_FinalProject-CurrentEDA/EDA_movies.csv", names=["movie", "id"])

In [8]:
movies_and_ids.head()

movie         id
0                           Tempus fugit  tt0390545
1                                  Sanam  tt0254749
2  Bike Week Exposed: Saints and Sinners  tt0367544
3                     Dancer in the Dark  tt0168629
4                     My Life Without Me  tt0314412

## Create IMDB object and test out connection

In [9]:
ia = imdb.IMDb()

In [10]:
s_result = ia.search_movie('The Princess Bride')

# Retrieves default information for the first result (a Movie object).
the_unt = s_result[0]
ia.update(the_unt)

# Print some information.
print the_unt['runtime']
print the_unt['rating']
director = the_unt['director'] # get a list of Person objects.


[u'98']
8.1


In [11]:
s2 = ia.get_movie(s_result[0].movieID)

In [12]:
movies_and_ids['id'][10]

'tt0040789'

## Function to extract data from IMDB

In [11]:
def get_data(offset, interval, q):
    all_results = np.array(['id','title', 'canonical title', 'kind', 'director', 'rating', 'year','votes', 
                         'mpaa','runtimes', 'color info', 'genres','Action', 
                         'Adventure', 'Adult', 'Animation', 'Comedy', 'Crime', 
                         'Documentary', 'Drama', 'Family', 'Fantasy', 'Film-Noir',
                         'Horror', 'Musical', 'Mystery', 'Romance', 'Sci-Fi', 'Short', 'Thriller',
                         'War', 'Western'])
                        
    all_genres = ['Action', 'Adventure', 'Adult', 'Animation',
    'Comedy', 'Crime', 'Documentary', 'Drama', 'Family', 'Fantasy', 'Film-Noir',
    'Horror', 'Musical', 'Mystery', 'Romance', 'Sci-Fi', 'Short', 'Thriller',
    'War', 'Western']

    for num,each in enumerate(movies_and_ids['id'][offset:offset+interval]):
        try:
            result = ia.get_movie(each[3:])
            if (num+1)%5==0: print num
            results = [each]
            for index, each in enumerate(['title', 'canonical title', 'kind', 
                                          'director', 'rating', 'year','votes', 
                                          'mpaa','runtimes', 'color info', 'genres']):
                try:
                    test = result[each]
                    if each in ['genres','director', 'runtimes']: results.append(str(test[0]))
                    else: results.append(str(test))
                except:
                    results.append("null")
            for index, each in enumerate(all_genres):
                try:
                    genre_result = result['genres']
                    if each in genre_result:
                        results.append(1)
                    else:
                        results.append(0)
                except:
                    results.append("null")
            all_results = np.vstack((all_results, results))
        except:
            continue
    features_df = pandas.DataFrame(data=all_results[1:,],  columns=all_results[0,])
    q.put(features_df)

## Use two processes to speed up the extraction

In [12]:
from multiprocessing import Process, Queue

if __name__ == '__main__':
    q = Queue()
    for offset in [0,1000]:
        interval = 1000
        Process(target=get_data, args=(offset,interval,q)).start()


4
4
9
9
14
14
19
19
24
24
29
34
29
39
34
44
39
44
49
49
54
54
59
59
64
64
69
69
74
74
79
79
84
84
89
89
94
94
99
99
104
104
109
109
114
114
119
119
124
124
129
129
134
134
139
139
144
144
149
149
154
154
159
159
164
164
169
169
174
174
179
179
184
184
189
189
194
194
199
199
204
204


2017-04-04 16:29:59,797 WARNING [imdbpy.parser.http.urlopener] /home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py:248: 404 code returned for //akas.imdb.com/title/tt0321320/combined: Not Found (headers: Date: Tue, 04 Apr 2017 20:49:15 GMT
Server: HTTPDaemon
X-Frame-Options: SAMEORIGIN
Cache-Control: private
Content-Type: text/html; charset=iso-8859-1
Set-Cookie: cache=BCYq3avOo5MArQbC310u5puZPH0xZIW80AZYEjFL5VQWaKT6sRTR4VlGLqhqhYU7FbX3XMRKTjfiYG4uTnrG5AzPPvAixNQhbv-PqtIga2bb_Nc;path=/;domain=.imdb.com
Set-Cookie: uu=BCYjV-og9cBSrO0oFjx34DRG1X4xZIW80AZYEjFL5VQWaKSQA4jDYDp4we0W1K7jU6pXzH_vvWk6H4RwCwdj6K7aWsr5Zn-TTV3Xi03PWfkCTbJ4YM0q-ay12FK7a59X9zle1J9V9CG3giZeEKnXUYv262vtlkbilE4gP0-Wsov0jD_UeX5u5-mXLWthRsRrMnFlloF5CJDzZy2BKImr8vl-tRFWNsJ70JlLHJOfvC2_C0p8XKVfrqugz3uYYNbkfmCA6W9AfWfdid6SSQCetuPnz2AvEZg0UXe1eTdeF71aeKzvh8Tu5djU-2rki7AYaV1KxGsXRtDjZXxepkwxCCmH4A;expires=Thu, 30 Dec 2037 00:00:00 GMT;path=/;domain=.imdb.com
Set-Cookie: cache=BCYutiQTx2EEEcdMq0_JnVHmA

209
209
214
214
219
219
224
224
229
229
234
234
239
239
244
244
249
249
254
254
259
259
264
264
269
269
274
274
279
279
284
289
284
289
294
294


2017-04-04 16:32:54,714 WARNING [imdbpy.parser.http.urlopener] /home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py:248: 404 code returned for //akas.imdb.com/title/tt0120781/combined: Not Found (headers: Date: Tue, 04 Apr 2017 20:52:10 GMT
Server: HTTPDaemon
X-Frame-Options: SAMEORIGIN
Cache-Control: private
Content-Type: text/html; charset=iso-8859-1
Set-Cookie: cache=BCYgxUC3JXy4Pf0smictCYLxDFyZG4TEpAw2T6EswYS54hoPW3ZjpiniaKxI1mPUQk1dLTd9g-kyQSbRs__VqO2w6M_0eSd03m8Pwu3N8Hp404E;path=/;domain=.imdb.com
Set-Cookie: uu=BCYjFu1ErfHJHtCJ4HJ-blbAN12ZG4TEpAw2T6EswYS54hoHR4tiYlYZyJfCobs9cg893HS4cPUaa_Wvx2EuNthW6zd6k3K7Ooqc76rhR43a5bu-p2OV41G6AOz2xEOMAyio4jvq1XvAePBsDXgVXj6OrdFRuhB627e4sYLWaME4qpy9B8oMKiDM3QPeA5Aqg-M5KUlaTLbci1wACcDFRHttPsVTtDYnNXe0lFLT7mLkimDCjb9Nj7VZ-ZI-F5l6MQ6d-bTatb3uXdPo5t1AUw0dtQUZV6iKfeRIIx_WGxDrLZDt7Ck-CF8Z2ypYZF6284I5uCONK-P7_QfbUVsQztlrpA;expires=Thu, 30 Dec 2037 00:00:00 GMT;path=/;domain=.imdb.com
Set-Cookie: cache=BCYp2YbwOXEBfoYCDLcqOH-17

299
304
299
309
304
309
314
319
314
324
319
329
334
324
339


2017-04-04 16:34:07,054 CRITICAL [imdbpy] /home/cs109b/.local/lib/python2.7/site-packages/imdb/_exceptions.py:35: IMDbParserError exception raised; args: ('invalid movieID "299858/": invalid literal for int() with base 10: \'299858/\'',); kwds: {}
Traceback (most recent call last):
  File "/home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py", line 344, in _normalize_movieID
    return '%07d' % int(movieID)
ValueError: invalid literal for int() with base 10: '299858/'
2017-04-04 16:34:07,656 WARNING [imdbpy.parser.http.urlopener] /home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py:248: 404 code returned for //akas.imdb.com/title/tt0942900/combined: Not Found (headers: Date: Tue, 04 Apr 2017 20:53:23 GMT
Server: HTTPDaemon
X-Frame-Options: SAMEORIGIN
Cache-Control: private
Content-Type: text/html; charset=iso-8859-1
Set-Cookie: cache=BCYnHA1zPljHZokz0M01LcFncLxKHXStlvF7t3I-GFuswKd4NCOA2GpkTO1iS7gi9DnhGCufKQwAMCu_6zvZrcLMAInrcMki7EzFi8oY3Q

329
344
334
349
354
339
359
344
364
349
369
374
354
359
379
364
384
369
374
389
379
394
384
399
389
404
394
409
399
404
414
409
419
414
424
419
429
424
434
429
439
434
439
444
444
449
449
454
459
454
459
464
464
469
469
474
474
479
479
484
484
489
489
494
494
499
499
504
504


2017-04-04 16:39:01,069 CRITICAL [imdbpy] /home/cs109b/.local/lib/python2.7/site-packages/imdb/_exceptions.py:35: IMDbDataAccessError exception raised; args: ({'exception type': 'IOError', 'url': 'http://akas.imdb.com/title/tt0238270/plotsummary', 'errcode': 'socket error', 'proxy': '', 'original exception': IOError('socket error', gaierror(-3, 'Temporary failure in name resolution')), 'errmsg': '[Errno -3] Temporary failure in name resolution'},); kwds: {}
Traceback (most recent call last):
  File "/home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py", line 202, in retrieve_unicode
    uopener = self.open(url)
  File "/usr/lib/python2.7/urllib.py", line 213, in open
    return getattr(self, name)(url)
  File "/usr/lib/python2.7/urllib.py", line 350, in open_http
    h.endheaders(data)
  File "/usr/lib/python2.7/httplib.py", line 1053, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python2.7/httplib.py", line 897, in _send_output
    self.sen

509
509
514
514
519
519
524
529
524
534
539
529
534
544


2017-04-04 16:40:05,543 WARNING [imdbpy.parser.http.urlopener] /home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py:248: 404 code returned for //akas.imdb.com/title/tt0122413/combined: Not Found (headers: Date: Tue, 04 Apr 2017 20:59:21 GMT
Server: HTTPDaemon
X-Frame-Options: SAMEORIGIN
Cache-Control: private
Content-Type: text/html; charset=iso-8859-1
Set-Cookie: cache=BCYq3TmSUNOe_Q3o7TOtiQD2qutiAIv0rBddbx0D-AZ2qNGBD1icx2BMCx_WRPq_mXVEXuc2BNu-aphZDT1DzGOeD2uD17uX-bUi03H9h-CgaUc;path=/;domain=.imdb.com
Set-Cookie: uu=BCYvydT-GY8LUVb26_NWNeC5lOxiAIv0rBddbx0D-AZ2qNEP2bNUVym9h4DRU3imeTudbPJ63Erf9mA-gf6y3b5iDZOEjlyONagNfw2P_8DOADoSkHuvehGfl9Fs8tFGxZt5hXq11MgkjtPPOUnwbpPR3uGhVqy0QQ_2vCko3xcPdsx5zlWchJ3iNC5WjH1LW9-JA2hHYjoo3jstr-WigSIU_zW28RdGPct2r6DhGTsC9OnPj-CP5hO_8LrmBJPnTJIEeduVqCDvPBR3O_GbChmvgxQ0DDMFHgPlBeHieLl1d_z37rX4CS-C29mIKVQ_siipfIAzPYk34fE05rhReQ26fQ;expires=Thu, 30 Dec 2037 00:00:00 GMT;path=/;domain=.imdb.com
Set-Cookie: cache=BCYtK9WVjmoFRIsoUWsUWe-YO

549
539
554
544
559
549
554
564
569
559
564
574
569
579
574
584
579
589
584
594
589
599
594
604
599
609
604
614
609
619
614
624
619
629
624
634
629
639
634
644
639
644
649
649
654
654
659
659
664
664
669
669
674
679
674
684
689
679
694
684
699
689
694


2017-04-04 16:44:31,230 CRITICAL [imdbpy] /home/cs109b/.local/lib/python2.7/site-packages/imdb/_exceptions.py:35: IMDbDataAccessError exception raised; args: ({'exception type': 'IOError', 'url': 'http://akas.imdb.com/title/tt0096860/plotsummary', 'errcode': 'socket error', 'proxy': '', 'original exception': IOError('socket error', gaierror(-3, 'Temporary failure in name resolution')), 'errmsg': '[Errno -3] Temporary failure in name resolution'},); kwds: {}
Traceback (most recent call last):
  File "/home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py", line 202, in retrieve_unicode
    uopener = self.open(url)
  File "/usr/lib/python2.7/urllib.py", line 213, in open
    return getattr(self, name)(url)
  File "/usr/lib/python2.7/urllib.py", line 350, in open_http
    h.endheaders(data)
  File "/usr/lib/python2.7/httplib.py", line 1053, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python2.7/httplib.py", line 897, in _send_output
    self.sen

704
699
704
709
709
714
719
714
719
724
724
729
729
734
734
739


2017-04-04 16:45:33,193 WARNING [imdbpy.parser.http.urlopener] /home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py:248: 404 code returned for //akas.imdb.com/title/tt0136917/combined: Not Found (headers: Date: Tue, 04 Apr 2017 21:04:48 GMT
Server: HTTPDaemon
X-Frame-Options: SAMEORIGIN
Cache-Control: private
Content-Type: text/html; charset=iso-8859-1
Set-Cookie: cache=BCYnIkFY6oIBMWRUAVSCU22l3MQfZBipVGI5zvFaJ06m77yInTF3YQcPu46JXka2W_WR1KQxZvKeCOiCJxxKRy4dMDv_g2398XmB3XF55NQ_11E;path=/;domain=.imdb.com
Set-Cookie: uu=BCYujo6V59L3CHR79onTWAclKcUfZBipVGI5zvFaJ06m77yhKCVmaKRWd8-Tcz-6xw78tHTClIAKxPg8u_CkMrQsI5q4eSHWczeQOJ8yLG_l9xllFar2z9UwDOZ0-59Aq6HU3grKuc2ajLSb3_ixFT1aBKgrvw-ZzD4U2-apqDubuRjrhh2Rt31ZUdG2YHTtpbouB4kXZYwNEIaZi8eqjMbOZybujzqkwq9QpIwKYGVAhw4KGq1plg-kmW9ujFtXlzmK5VjKqM0C-ZKFOdcFtSo7Za9nZ-dkh3gt3liOwsCXTrdnADUfEGVY5OMaACpi4qJotgauCvZ-mpFzYGKGGgs5qg;expires=Thu, 30 Dec 2037 00:00:00 GMT;path=/;domain=.imdb.com
Set-Cookie: cache=BCYpePHYjbqsRlElTNIo_0Byp

739
744
744
749
749
754
754
759
759
764
764
769
769
774
774
779
779
784
784
789
789
794
794
799
799
804
804
809
809
814
814
819
819
824
824
829
829
834
834
839


2017-04-04 16:48:14,317 WARNING [imdbpy.parser.http.urlopener] /home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py:248: 404 code returned for //akas.imdb.com/title/tt0987018/combined: Not Found (headers: Date: Tue, 04 Apr 2017 21:07:29 GMT
Server: HTTPDaemon
X-Frame-Options: SAMEORIGIN
Cache-Control: private
Content-Type: text/html; charset=iso-8859-1
Set-Cookie: cache=BCYpWnnAUMHtP0GIM3lMVkkVerkTQVIWizey6E9VmI2ZRNTsNlhw-PaiWYl3n8ANqqFK8l25QCh9UA4cyK68NCvUOzAkDX10cDY30HNQg_s_vD8;path=/;domain=.imdb.com
Set-Cookie: uu=BCYuBpiyKDdtj5Yr4f2YXybPEboTQVIWizey6E9VmI2ZRNSf2q9xI8_B67I0OvZrivYm7L6OWOvIWsIj5RSiBOg56sOi_Yqiyg660pjWPQ1_tjnTUxNjU6fuAUSww2h_fR92vsjDhNgQraVbR3iEFCqbadHB5RXQrITnM1Lz9DO0DzkJYGOPjTmh6V6vbaDf2NwuBomZEmiHJ78KMkPEza-03MwB162_Q7CZo2oygU2EZc_LfSHv4MVO7VaD8oWZKK7U69nOUIGmqsBYZpFGfgrwLwjWMJqG1kPuDD4COHJyGAs_hNFPfufCD4w5PM5EXcGMgxq-E8N6RXweS1JYVPviAg;expires=Thu, 30 Dec 2037 00:00:00 GMT;path=/;domain=.imdb.com
Set-Cookie: cache=BCYsE4VgEX5TWH8F9fKZW3dAE

839
844
849
844
854
849
859


2017-04-04 16:48:39,364 WARNING [imdbpy.parser.http.urlopener] /home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py:248: 404 code returned for //akas.imdb.com/title/tt0808340/combined: Not Found (headers: Date: Tue, 04 Apr 2017 21:07:54 GMT
Server: HTTPDaemon
X-Frame-Options: SAMEORIGIN
Cache-Control: private
Content-Type: text/html; charset=iso-8859-1
Set-Cookie: cache=BCYqt4fEtsT5ThxE1JO_cT_DamfA9IR5iOZpY4GAiZbFkbRalpXqdY-xmdHlmnJBGUxXwkgQgu2h-yyXB7EeYx7udAbXbFtEgBbtEwyt3iw8MjY;path=/;domain=.imdb.com
Set-Cookie: uu=BCYtmsgKgYnKxa3yxgzv6qci4GjA9IR5iOZpY4GAiZbFkbT15DqLk-yr5asyd_khlmfF3ZPSS5_EXfwZqt5qV1sTbMzUfCyfFD-tAYNzGSntDJSe2daBUjngDaKz7TD1qU27ZjcyppU1Ox1GsLpsBDJMF1c3AIwTaQ5lFbBimKm-MVB-YpPKVPxfcr3reVeP3It26ca18c439M6zoxmf7q-VUNeDPlmFN-AGZMOVFl-qH6sG8VKKR63-RJDj_HTmBJ7HzhsaRPL9c74NDLoWhA2XbqadWt3Wb4JBG60oHaipPGBwJrwPk3ivzzj5liq3rmA3d-oTLjx3VgkO2FHuf_Zpdg;expires=Thu, 30 Dec 2037 00:00:00 GMT;path=/;domain=.imdb.com
Set-Cookie: cache=BCYs2_ozcicZlOEdMdc7BHuQh

854
864
859


2017-04-04 16:48:47,488 WARNING [imdbpy.parser.http.urlopener] /home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py:248: 404 code returned for //akas.imdb.com/title/tt0168854/combined: Not Found (headers: Date: Tue, 04 Apr 2017 21:08:02 GMT
Server: HTTPDaemon
X-Frame-Options: SAMEORIGIN
Cache-Control: private
Content-Type: text/html; charset=iso-8859-1
Set-Cookie: cache=BCYsbY6_CA6KkknJ3jzMD-hdz8psflnUhjFlmwVLTlQKCU2pOD2wQiFkhhk3lKwiqUa9_mVbK7U7LoSXTxL6RdC4q-S8Uo-rWr7vCnJB_L6oGQU;path=/;domain=.imdb.com
Set-Cookie: uu=BCYl4AtbewXKY-vklg_ur1rpIstsflnUhjFlmwVLTlQKCU3VDKhoK4oqcbV89947GvqoCbcTdjl__r9rhXymND8HP_wvxcbxni2DeTdsBxtwYnK03VMS_6EJbWaONAhXYAiU2xzMQdGifQBxdoLfB1ppBr4AL4xSL1tlVz22pIHFk7WYeGt_LYHhOdx4AbaILjeymZro3pIwV4dqx3wDPdcr8IC0Uw5x229xP_Ckf4GR0VOyS-48EJx1j7CkyaC7hg9MRb7SEnvOhvgh2Up2kkUhOEZI2Jvr57c8XmIUUbuGoOJLbc5UG-uYF62NM6Ehml4iNjY7YmqomyzeLsR-loO1SQ;expires=Thu, 30 Dec 2037 00:00:00 GMT;path=/;domain=.imdb.com
Set-Cookie: cache=BCYvNjaIVjllfu9VabbPD4c7g

864
869
869
874
874
879
879
884
884
889
894
889
899
894
904
899
909
914
904
919
909
914
919
924
924
929
929
934
934
939
939
944
944
949
949
954


2017-04-04 16:51:13,290 WARNING [imdbpy.parser.http.urlopener] /home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py:248: 404 code returned for //akas.imdb.com/title/tt0157654/combined: Not Found (headers: Date: Tue, 04 Apr 2017 21:10:28 GMT
Server: HTTPDaemon
X-Frame-Options: SAMEORIGIN
Cache-Control: private
Content-Type: text/html; charset=iso-8859-1
Set-Cookie: cache=BCYio0DEJVT-fbVlr_QVeV-PF0O94QotSKB1J24GJFi6RwMdTq-wFLBlmqIlGNhJH7UKsYNF5LhZMPEV_OIgnhIf7aIANUTAt0A9_dGzbQ3DgIw;path=/;domain=.imdb.com
Set-Cookie: uu=BCYmvl8ECt01kyLmNNfYdzmbJkS94QotSKB1J24GJFi6RwPKYbD_0vDAO02z_azsKVHVb-mG5uAV7ikSOeCUoEnFxEtyYEnIp_fTltYqxy5WhpSCRlysp1-zNcZt3Alw4g6u8km3vFZLVWMQGY1zIcdpvN2acSGiX1Cipzos9UQ7qNs3JRV9-szEG4MNXAkQdgnoyYz690DcmjBCjB3VKzvgfJ2HDsR3h9eMaRhvaZ67AV3P500ff0zmI-ga_i03b8N-2nFPXsezwmEZbBxSBT3ID_cgXRzLHd4zyENgdMeEBnr5EDptFx1SibyInwgLT0sUGkJ6pHS9wt5IWIUQYdcZ7w;expires=Thu, 30 Dec 2037 00:00:00 GMT;path=/;domain=.imdb.com
Set-Cookie: cache=BCYuCZLr63s7hCmwvXvf7QqP7

954


2017-04-04 16:51:13,422 WARNING [imdbpy.parser.http.urlopener] /home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py:248: 404 code returned for //akas.imdb.com/title/tt0157654/plotsummary: Not Found (headers: Date: Tue, 04 Apr 2017 21:10:28 GMT
Server: Server
X-Frame-Options: SAMEORIGIN
Content-Security-Policy: frame-ancestors 'self' imdb.com *.imdb.com *.media-imdb.com withoutabox.com *.withoutabox.com amazon.com *.amazon.com amazon.co.uk *.amazon.co.uk amazon.de *.amazon.de translate.google.com images.google.com www.google.com www.google.co.uk search.aol.com bing.com www.bing.com
Content-Type: text/html;charset=UTF-8
Content-Language: en-US
Vary: Accept-Encoding,User-Agent
P3P: policyref="http://i.imdb.com/images/p3p.xml",CP="CAO DSP LAW CUR ADM IVAo IVDo CONo OTPo OUR DELi PUBi OTRi BUS PHY ONL UNI PUR FIN COM NAV INT DEM CNT STA HEA PRE LOC GOV OTC "
Connection: close
)


959
959
964
964
969
969


2017-04-04 16:51:34,000 WARNING [imdbpy.parser.http.urlopener] /home/cs109b/.local/lib/python2.7/site-packages/imdb/parser/http/__init__.py:248: 404 code returned for //akas.imdb.com/title/tt0236968/combined: Not Found (headers: Date: Tue, 04 Apr 2017 21:10:49 GMT
Server: HTTPDaemon
X-Frame-Options: SAMEORIGIN
Cache-Control: private
Content-Type: text/html; charset=iso-8859-1
Set-Cookie: cache=BCYsogAD4sXeCdJmgj2wLonL_BVROd_JzlOFsmTys5NmVqNl_ZTkHtnmezrmCJa__TmCGLSSEPbNiy_pHoNnNz4puJgqAXoypZ6Z_GAJxespWmQ;path=/;domain=.imdb.com
Set-Cookie: uu=BCYvCG5_a5c5ql5tu_O9IEY5PRZROd_JzlOFsmTys5NmVqO1Czpvqu94WV65X-cIe4YzJ36-ixw_n_JKuWbu3hi080LoLyjXnc81Paduxu2eG2sWjfx0xQRbEGgCIixd-spBhTDW_ebMZbwPLdrOy4dS6uCq-Cq3XzH4f5Grpe-oVRba7uHq6nV8wzHEj4D8aR5Wlib2wLMsnS6nBtfaxcjLFpk_glwCqWsANjtQOWo2JKMM7TR6tCtTNp7KgfDjk4UuuoNlxqok8edGcJUiM6fYNKw_mwWw2s_wDRpgTNKRXy53J8N3pcCVzwemg9aLMzyRfBB7PH0mfmVIpGkf5L8JRQ;expires=Thu, 30 Dec 2037 00:00:00 GMT;path=/;domain=.imdb.com
Set-Cookie: cache=BCYim0RWAYxv16Vxaqh5Nx3Ap

974
974
979
979
984
984
989
989
994
999
994
999


## Collect dataframes from the q and combine

In [13]:
all_dfs = []
for _ in range(2):
    all_dfs.append(q.get())

In [15]:
output_result = all_dfs[0]
for each in all_dfs[1:]:
    output_result = output_result.append(each,ignore_index=True)

## Check data and export to csv

In [16]:
output_result.head()

id              title     canonical title         kind  \
0  tt0179099               null                null        movie   
1  tt0402362             Margam              Margam        movie   
2  tt0156075  The Shrunken City  Shrunken City, The  video movie   
3  tt4094088   Sweet as You Are    Sweet as You Are      episode   
4  tt0494546     The Ranch Girl     Ranch Girl, The        movie   

                  director rating  year votes  \
0  Haroldo Marinho Barbosa    5.5  1986    43   
1     Rajiv Vijayaraghavan    8.4  2003    11   
2             Ted Nicolaou    4.5  1998   163   
3              Angela Pope    7.2  1988    53   
4               Allan Dwan   null  1911  null   

                                   mpaa   runtimes   ...   Film-Noir Horror  \
0                                  null        110   ...           0      0   
1                                  null  India:108   ...        null   null   
2  Rated PG for fantasy action violence     USA:90   ...           0      0   
3                                  null       null   ...           0      0   
4                                  null       null   ...           0      0   

  Musical Mystery Romance Sci-Fi Short Thriller   War Western  
0       0       0       0      0     0        0     0       0  
1    null    null    null   null  null     null  null    null  
2       0       0       0      1     0        0     0       0  
3       0       0       0      0     0        0     0       0  
4       0       0       0      0     1        0     0       1  

[5 rows x 32 columns]

In [22]:
output_result.to_csv("imdb_features_sample_2.csv")

### possible keys we can get from IMDB
title; string; the "usual" title of the movie, like "The Untouchables".
long imdb title; string; "Uncommon Valor (1983/II) (TV)"
canonical title; string; the title in the canonical format,
                         like "Untouchables, The".
long imdb canonical title; string; "Patriot, The (2000)".
year; string; the year of release or '????' if unknown.
kind; string; one in ('movie', 'tv series', 'tv mini series', 'video game',
                      'video movie', 'tv movie', 'episode')
imdbIndex; string; the roman number for movies with the same title/year.
director; Person list; a list of director's name (e.g.: ['Brian De Palma'])
cast; Person list; list of actor/actress, with the currentRole instance
                   variable set to a Character object which describe his
                   role/duty.
cover url; string; the link to the image of the poster.
writer; Person list; list of writers ['Oscar Fraley (novel)']
plot; list; list of plots and authors of the plot.
rating; string; user rating on IMDb from 1 to 10 (e.g. '7.8')
votes; string; number of votes (e.g. '24,101')
runtimes; string list; in minutes ['119'] or something like ['USA:118',
          'UK:116']
number of episodes; int; number or episodes for a series.
color info; string list; ["Color (Technicolor)"]
countries; string list; production's country ['USA', 'Italy']
genres; string list; one or more in (Action, Adventure, Adult, Animation,
		Comedy, Crime, Documentary, Drama, Family, Fantasy, Film-Noir,
		Horror, Musical, Mystery, Romance, Sci-Fi, Short, Thriller,
		War, Western) and other genres defined by IMDb.
akas; string list; list of aka for this movie
languages; string list; list of languages
certificates; string list; ['UK:15', 'USA:R']
mpaa; string; the mpaa rating
episodes (series only); dictionary of dictionary; one key for every season,
                        one key for every episode in the season.
number of episodes (series only); int; total number of episodes.
number of seasons (series only); int; total number of seasons.
series years (series only); string; range of years when the series was produced.
episode of (episode only); Movie object; the parent series for an episode.
season (episode only); int; the season number.
episode (episode only); int; the number of the episode in the season.
long imdb episode title (episode only); string; episode and series title.
series title; string.
canonical series title; string.